src/_P145_MQxxx.ino

```diff
@@ -0,0 +1,281 @@
+      Device[deviceCount].InverseLogicOption = false;
+      Device[deviceCount].FormulaOption = true;
+      Device[deviceCount].ValueCount = 1;
+      Device[deviceCount].SendDataOption = true;
+      Device[deviceCount].TimerOption = true;
+      Device[deviceCount].GlobalSyncOption = true;
+      break;
+    }
+    
+    case PLUGIN_GET_DEVICENAME:
+    {
+      string = F(PLUGIN_NAME_145);
+      break;
+    }
+
+    case PLUGIN_GET_DEVICEVALUENAMES:
+    {
+      strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[0], PSTR(PLUGIN_VALUENAME1_145));
+      break;
+    }
+
+    case PLUGIN_WEBFORM_LOAD:
+    {
+      // FormSelector with all predefined "Sensor - Gas" options
+      String options[P145_MAXTYPES] = {};
+      int optionValues[P145_MAXTYPES] = {};
+      int x = P145_data_struct::getNbrOfTypes();
+      if (x > P145_MAXTYPES) 
+      {
+        x = P145_MAXTYPES;    // Clip to prevent array boundary out of range access
+      }
+      for (int i=0; i<x; i++)
+      {
+        options[i] = concat(P145_data_struct::getTypeName(i), F(" - ")) + (String)P145_data_struct::getGasName(i);
+        optionValues[i] = i; 
+      }
+      addFormSelector(F("Sensor type"), F("plugin_145_sensor"), x, options, optionValues, P145_PCONFIG_SENSORT);
+
+# ifdef ESP32
+      // Analog input selection
+      addRowLabel(F("Analog Pin"));
+      addADC_PinSelect(AdcPinSelectPurpose::ADC_Touch_HallEffect, F("taskdevicepin1"), CONFIG_PIN1);
+# endif // ifdef ESP32
+
+      addFormFloatNumberBox(F("Load Resistance"), F("plugin_145_RLOAD"), P145_PCONFIG_RLOAD, 0.0, 10e6, 2U);
+      addUnit(F("Ohm"));
+      addFormFloatNumberBox(F("R Zero"), F("plugin_145_RZERO"), P145_PCONFIG_RZERO, 0.0, 10e6, 2U);
+      addUnit(F("Ohm"));
+       addFormFloatNumberBox(F("Reference Level"), F("plugin_145_REFLEVEL"), P145_PCONFIG_REF, 0.0, 10e6, 2U);
+      addUnit(F("ppm"));
+      P145_data_struct *P145_data = static_cast<P145_data_struct *>(getPluginTaskData(event->TaskIndex));
+      if (P145_data != nullptr)
+      {
+        float calVal = P145_data->getCalibrationValue();
+        if (calVal > 0.0) 
+        {
+          addFormNote((String)F("Current measurement suggests Rzero= ") + (String)calVal);
+        }
+      }
+
+      addFormSeparator(2);
+      // Auto calibration and Temp/hum compensation flags are bitfields in P145_PCONFIG_FLAGS
+      bool compensate = P145_PCONFIG_FLAGS & 0x0001;
+      bool calibrate = (P145_PCONFIG_FLAGS >> 1) & 0x0001;
+      addFormCheckBox(F("Enable automatic calibration"), F("plugin_145_enable_calibrarion"), calibrate);
+      addFormCheckBox(F("Enable temp/humid compensation"), F("plugin_145_enable_compensation"), compensate);
+      addFormNote(F("If this is enabled, the Temperature and Humidity values below need to be configured."));
+      // temperature
+      addHtml(F("<TR><TD>Temperature:<TD>"));
+      addTaskSelect(F("plugin_145_temperature_task"), P145_PCONFIG_TEMP_TASK);
```

Comment:
Yesterday, I made a change to the `addTaskSelect` function.
It now includes a `Not Set` option, which has an invalid TaskIndex_t value.
So right after this line, you chould check using `if (validTaskIndex(P145_PCONFIG_TEMP_MAX))` to see if you should continue where this taskindex is being used.

Reply:
Done. Question: I see that the form is reloaded when another task is selected. Is this also possible if a checkbox is changed? I want to skip the compensation task/value settings when no compensation is selected.

Reply:
I am already working on adding some extras on this `addTaskSelect` function to allow it to still reload and update the settings, but not (yet) saving it.
But it does have some side-effects, so I'm not 100% that will remain.

What do you want to do with it?
We can also add some JavaScript code to hide some elements.
For this I have added some functions which will specifically add a HTML id to wrappers so you can hide an entire row (or multiple rows) on change of some combo box. (a checkbox can also be done as a combo box)
I've done this for C018 and for the ESPEasySerial webforms functions.